src/main/java/placement/model/board/Board.java

```diff
@@ -68,80 +64,40 @@ public int cols() {
-    }
 
-    /**
-     * Places given ship on the board if it's possible.
-     *
-     * @param ship        ship to place.
-     * @param coordinates coordinates of the ship's "head".
-     * @see ShipOnBufferException
-     * @see ShipOnOccupiedFieldException
-     * @see ShipOutOfBoardException
-     */
-    public void placeShip(Ship ship, Coordinates coordinates) {
-        try {
-            Set<Coordinates> mastCoordinates = getMastCoordinates(ship, coordinates);
-            mastCoordinates.forEach(coordinate -> {
-                boardFields.markAsOccupied(coordinate);
-                surroundWithBuffer(coordinate);
-                ship.addPositionCoord(coordinate);
-            });
-            ship.markAsPlaced();
-
-        } catch (ShipOutOfBoardException | ShipOnBufferException | ShipOnOccupiedFieldException e) {
-            System.err.println(e.getMessage());
-        }
+    public boolean isMastPlaceable(Coordinates mastCoordinates) {
+        Field field = boardFields.getField(mastCoordinates);
+        return field.isPlaceable();
     }
 
-    public void placeShip(Ship ship, int x, int y) {
-        Coordinates coordinates = new Coordinates(x, y);
-        placeShip(ship, coordinates);
-    }
-
-    private Set<Coordinates> getMastCoordinates(Ship ship, Coordinates coordinates)
-            throws ShipOutOfBoardException, ShipOnBufferException, ShipOnOccupiedFieldException {
-        Set<Coordinates> mastCoordinates = new HashSet<>();
-        while (mastCoordinates.size() < ship.getMastNumber()) {
-            checkIsAbleToPlace(coordinates);
-            Coordinates nextCoord = ship.nextCoordinates();
-            mastCoordinates.add(coordinates);
-            coordinates = coordinates.add(nextCoord);
-        }
-        return mastCoordinates;
-    }
-
-    private void checkIsAbleToPlace(Coordinates coordinates)
-            throws ShipOutOfBoardException, ShipOnBufferException, ShipOnOccupiedFieldException {
-        Field field = boardFields.getField(coordinates);
-        field.checkState();
+    public void placeMast(Coordinates coordinates) {
+        boardFields.markAsOccupied(coordinates);
+        surroundWithBuffer(coordinates);
     }
 
     private void surroundWithBuffer(Coordinates mastCoordinates) {
         Set<Coordinates> neighbours = getNeighboursToMarkAsBuffer(mastCoordinates);
         neighbours.forEach(boardFields::markAsBuffer);
     }
 
-    private Set<Coordinates> getNeighboursToMarkAsBuffer(Coordinates coordinates) {
+    private Set<Coordinates> getNeighboursToMarkAsBuffer(Coordinates coordinatesToSurrounded) {
         HashSet<Coordinates> neighbours = new HashSet<>();
         for (int i = -1; i <= 1; i++) {
             for (int j = -1; j <= 1; j++) {
                 Coordinates neighbour = new Coordinates(
-                        coordinates.getX() + i,
-                        coordinates.getY() + j);
+                        coordinatesToSurrounded.getX() + i,
```

Comment:
You have already a method to add coordinates to each other (in Coordinates class)